helios-services/src/test/java/com/spotify/helios/master/reaper/OldJobReaperTest.java

```diff
@@ -126,55 +135,176 @@ public JobStatus getJobStatus() {
+        name.getMethodName(),
+        deployments(JobId.fromString(name.getMethodName()), 2),
+        emptyList());
+    testReap(datapoint, masterModel, false);
+  }
+
+  /**
+   * A job deployed, with history, and last used too long ago should NOT BE reaped.
+   */
+  @Test
+  public void jobDeployedWithHistoryLastUsedTooLongAgoNotReaped() throws Exception {
+    final MasterModel masterModel = mock(MasterModel.class);
+    final Datapoint datapoint = new Datapoint(
+        name.getMethodName(),
+        deployments(JobId.fromString(name.getMethodName()), 3),
+        events(ImmutableList.of(HOURS.toMillis(20), HOURS.toMillis(22))));
+    testReap(datapoint, masterModel, false);
+  }
+
+  /**
+   * A job deployed, with history, and last used recently should NOT BE reaped.
+   */
+  @Test
+  public void jobDeployedWithHistoryLastUsedRecentlyNotReaped() throws Exception {
+    final MasterModel masterModel = mock(MasterModel.class);
+    final Datapoint datapoint = new Datapoint(
+        name.getMethodName(),
+        deployments(JobId.fromString(name.getMethodName()), 3),
+        events(ImmutableList.of(HOURS.toMillis(20), HOURS.toMillis(40))));
+    testReap(datapoint, masterModel, false);
+  }
+
+  /**
+   * A job not deployed, with history, last used too long ago and part of a deployment group
+   * should NOT BE reaped.
+   */
+  @Test
+  public void jobDeployedWithHistoryLastUsedTooLongAgoInDGNotReaped() throws Exception {
+    final MasterModel masterModel = mock(MasterModel.class);
+
+    when(masterModel.getDeploymentGroups())
+        .thenReturn(ImmutableMap.of(
+            "testdg",
+            new DeploymentGroup("name", new ArrayList<>(), JobId.fromString(name.getMethodName()),
+                RolloutOptions.getDefault(), DeploymentGroup.RollingUpdateReason.MANUAL)));
+
+    final Datapoint datapoint = new Datapoint(
+        name.getMethodName(),
+        emptyMap(),
+        events(ImmutableList.of(HOURS.toMillis(2), HOURS.toMillis(3))));
+
+    testReap(datapoint, masterModel, false);
+  }
+
+  /**
+   * A job not deployed, with history, last used too long ago and part of a deployment group
+   * should BE reaped.
+   */
   @Test
-  public void testOldJobReaper() throws Exception {
+  public void jobDeployedWithHistoryLastUsedTooLongAgoNotInDGReaped() throws Exception {
     final MasterModel masterModel = mock(MasterModel.class);
+
+    when(masterModel.getDeploymentGroups())
+        .thenReturn(ImmutableMap.of(
+            "testdg",
+            new DeploymentGroup("name", new ArrayList<>(), JobId.fromString("framazama"),
+                RolloutOptions.getDefault(), DeploymentGroup.RollingUpdateReason.MANUAL)));
+
+    final Datapoint datapoint = new Datapoint(
+        name.getMethodName(),
+        emptyMap(),
+        events(ImmutableList.of(HOURS.toMillis(2), HOURS.toMillis(3))));
+
+    testReap(datapoint, masterModel, true);
+  }
+
+  private void testReap(
+      final Datapoint datapoint,
```

Comment:
with this refactoring of the test, the `Datapoint` class seems a bit less useful, I wonder if it would just be simpler to pass the data in Datapoint's fields into this method as parameters?

Reply:
👍 absolutely